asabiris/orchestration/sendslack.py

```diff
@@ -44,58 +47,50 @@ async def send_to_slack(self, msg):
 
-		# templates must be stores in /Templates/Slack
-		if not template.startswith("/Templates/Slack/"):
-			raise PathError(use_case='Slack', invalid_path=template)
-
-		params = body.get("params", {})
-		output = await self.JinjaService.format(template, params)
-
-		if attachments is None:
-			# No attachments provided, send the message as a block
-
-			if template.endswith('.md'):
-				# Translate output from markdown to plain text
-				fallback_message = self.MarkdownFormatterService.unformat(output)
-
-				# See https://api.slack.com/reference/block-kit/blocks
-				blocks = [
-					{
-						"type": "section",
-						"text": {
-							"type": "mrkdwn",
-							"text": output
+		try:
+			# templates must be stores in /Templates/Slack
+			if not template.startswith("/Templates/Slack/"):
+				raise PathError(use_case='Slack', invalid_path=template)
+
+			params = body.get("params", {})
+			output = await self.JinjaService.format(template, params)
+
+			if attachments is None:
+				# No attachments provided, send the message as a block
+
+				if template.endswith('.md'):
+					# Translate output from markdown to plain text
+					fallback_message = self.MarkdownFormatterService.unformat(output)
+
+					# See https://api.slack.com/reference/block-kit/blocks
+					blocks = [
+						{
+							"type": "section",
+							"text": {
+								"type": "mrkdwn",
+								"text": output
+							}
 						}
-					}
-				]
-
-			else:
-				# This is a plain text Slack message
-				fallback_message = output
-				blocks = None
-
-			await self.SlackOutputService.send_message(blocks, fallback_message)
-			return
-
-		# Sending attachments
-
-		output = self.MarkdownFormatterService.unformat(output)
-		atts_gen = self.AttachmentRenderingService.render_attachment(attachments)
-		await self.SlackOutputService.send_files(output, atts_gen)
+					]
 
+				else:
+					# This is a plain text Slack message
+					fallback_message = output
+					blocks = None
 
-	async def render_attachment(self, template, params):
-		"""
-		This method renders attachment based on the depending on the extension of template.
-		"""
+				await self.SlackOutputService.send_message(blocks, fallback_message)
+				return
 
-		try:
-			jinja_output = await self.JinjaService.format(template, params)
-		except KeyError:
-			L.warning("Failed to load or render a template (missing?)", struct_data={'template': template})
-			raise
+			# Sending attachments
 
-		return jinja_output
+			output = self.MarkdownFormatterService.unformat(output)
+			atts_gen = self.AttachmentRenderingService.render_attachment(attachments)
+			await self.SlackOutputService.send_files(output, atts_gen)
+		except Exception as e:
+			L.exception("Error occurred when preparing slack notification")
+			error_message = self._generate_error_message_slack(str(e))
+			blocks = None
+			await self.SlackOutputService.send_message(blocks, error_message)
```

Comment:
what if the exception comes from sending the message? then, you send another message which does not reach its target and you return 200 in response?

That's not a good approach